File: Source/aoFormWizard3/Models/Db/LibraryFolderModel.vb


Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses

Namespace Models.Db
    Public Class LibraryFolderModel        '<------ set set model Name and everywhere that matches this string
        Inherits BaseModel
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "Library Folders"      '<------ set content name
        Public Const contentTableName As String = "ccLibraryFolders"   '<------ set to tablename for the primary content (used for cache names)
        Private Shadows Const contentDataSource As String = "default"             '<------ set to datasource if not default
        '
        '====================================================================================================
        ' -- instance properties
        'instancePropertiesGoHere
        Public Property description As String
        Public Property parentid As Integer
        '
        '====================================================================================================
        Public Overloads Shared Function add(cp As CPBaseClass) As LibraryFolderModel
            Return add(Of LibraryFolderModel)(cp)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cp As CPBaseClass, recordId As Integer) As LibraryFolderModel
            Return create(Of LibraryFolderModel)(cp, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cp As CPBaseClass, recordGuid As String) As LibraryFolderModel
            Return create(Of LibraryFolderModel)(cp, recordGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cp As CPBaseClass, recordName As String) As LibraryFolderModel
            Return createByName(Of LibraryFolderModel)(cp, recordName)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub save(cp As CPBaseClass)
            MyBase.save(cp)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cp As CPBaseClass, recordId As Integer)
            delete(Of LibraryFolderModel)(cp, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cp As CPBaseClass, ccGuid As String)
            delete(Of LibraryFolderModel)(cp, ccGuid)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cp As CPBaseClass, sqlCriteria As String, Optional sqlOrderBy As String = "id") As List(Of LibraryFolderModel)
            Return createList(Of LibraryFolderModel)(cp, sqlCriteria, sqlOrderBy)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cp As CPBaseClass, recordId As Integer) As String
            Return BaseModel.getRecordName(Of LibraryFolderModel)(cp, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cp As CPBaseClass, ccGuid As String) As String
            Return BaseModel.getRecordName(Of LibraryFolderModel)(cp, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordId(cp As CPBaseClass, ccGuid As String) As Integer
            Return BaseModel.getRecordId(Of LibraryFolderModel)(cp, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getCount(cp As CPBaseClass, sqlCriteria As String) As Integer
            Return BaseModel.getCount(Of LibraryFolderModel)(cp, sqlCriteria)
        End Function
        '
        '====================================================================================================
        Public Overloads Function getUploadPath(fieldName As String) As String
            Return MyBase.getUploadPath(Of LibraryFolderModel)(fieldName)
        End Function

    End Class
End Namespace

File: Source/aoFormWizard3/Models/Db/LibraryFolderModel.cs

//using System.Collections.Generic;
//using Contensive.BaseClasses;

//namespace Contensive.Addon.aoFormWizard3.Models.Db {
//    public class LibraryFolderModel : BaseModel {        // <------ set set model Name and everywhere that matches this string
//        // 
//        // ====================================================================================================
//        // -- const
//        public const string contentName = "Library Folders";      // <------ set content name
//        public const string contentTableName = "ccLibraryFolders";   // <------ set to tablename for the primary content (used for cache names)
//        private new const string contentDataSource = "default";             // <------ set to datasource if not default
//        // 
//        // ====================================================================================================
//        // -- instance properties
//        // instancePropertiesGoHere
//        public string description { get; set; }
//        public int parentid { get; set; }
//        // 
//        // ====================================================================================================
//        public static LibraryFolderModel @add(CPBaseClass cp) {
//            return @add<LibraryFolderModel>(cp);
//        }
//        // 
//        // ====================================================================================================
//        public static LibraryFolderModel create(CPBaseClass cp, int recordId) {
//            return create<LibraryFolderModel>(cp, recordId);
//        }
//        // 
//        // ====================================================================================================
//        public static LibraryFolderModel create(CPBaseClass cp, string recordGuid) {
//            return create<LibraryFolderModel>(cp, recordGuid);
//        }
//        // 
//        // ====================================================================================================
//        public static LibraryFolderModel createByName(CPBaseClass cp, string recordName) {
//            return createByName<LibraryFolderModel>(cp, recordName);
//        }
//        // 
//        // ====================================================================================================
//        public new void save(CPBaseClass cp) {
//            base.save(cp);
//        }
//        // 
//        // ====================================================================================================
//        public static void delete(CPBaseClass cp, int recordId) {
//            delete<LibraryFolderModel>(cp, recordId);
//        }
//        // 
//        // ====================================================================================================
//        public static void delete(CPBaseClass cp, string ccGuid) {
//            delete<LibraryFolderModel>(cp, ccGuid);
//        }
//        // 
//        // ====================================================================================================
//        public static List<LibraryFolderModel> createList(CPBaseClass cp, string sqlCriteria, string sqlOrderBy = "id") {
//            return createList<LibraryFolderModel>(cp, sqlCriteria, sqlOrderBy);
//        }
//        // 
//        // ====================================================================================================
//        public static string getRecordName(CPBaseClass cp, int recordId) {
//            return getRecordName<LibraryFolderModel>(cp, recordId);
//        }
//        // 
//        // ====================================================================================================
//        public static string getRecordName(CPBaseClass cp, string ccGuid) {
//            return getRecordName<LibraryFolderModel>(cp, ccGuid);
//        }
//        // 
//        // ====================================================================================================
//        public static int getRecordId(CPBaseClass cp, string ccGuid) {
//            return getRecordId<LibraryFolderModel>(cp, ccGuid);
//        }
//        // 
//        // ====================================================================================================
//        public static int getCount(CPBaseClass cp, string sqlCriteria) {
//            return getCount<LibraryFolderModel>(cp, sqlCriteria);
//        }
//        // 
//        // ====================================================================================================
//        public string getUploadPath(string fieldName) {
//            return getUploadPath<LibraryFolderModel>(fieldName);
//        }

//    }
//}